From 50eea4c2998f9f751502bc51d8fa4c6218cfc329 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:38:06 -0400
Subject: [alsa] ice1712: Correcting/completing #defines for REGS

Message-id: <140972823573-377-patchset-jkysela@redhat.com>
Patchwork-id: 91073
O-Subject: [RHEL 7.1 PATCH 376/442] ALSA: ice1712: Correcting/completing #defines for REGS
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This small patch completes #defines for Control/Status Register,
adds comments for the missing ones there and on the Interrupt Mask
Register and additionally corrects "#define ICE1712_SERR_LEVEL 0x04 -> 0x08",
according to documentation.

Signed-off-by: Konstantinos Tsimpoukas <kostaslinuxxx@gmail.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 2f6ba2b31ba9442d2a16aa2a7e1b4116de2de14f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/ice1712/ice1712.h b/sound/pci/ice1712/ice1712.h
index b209fc3..58f8f2a 100644
--- a/sound/pci/ice1712/ice1712.h
+++ b/sound/pci/ice1712/ice1712.h
@@ -41,14 +41,17 @@
 #define ICEREG(ice, x) ((ice)->port + ICE1712_REG_##x)
 
 #define ICE1712_REG_CONTROL  0x00 /* byte */
-#define   ICE1712_RESET   0x80 /* reset whole chip */
-#define   ICE1712_SERR_LEVEL  0x04 /* SERR# level otherwise edge */
+#define   ICE1712_RESET   0x80 /* soft reset whole chip */
+#define   ICE1712_SERR_ASSERT_DS_DMA 0x40    /* disabled SERR# assertion for the DS DMA Ch-C irq otherwise enabled */
+#define   ICE1712_DOS_VOL  0x10    /* DOS WT/FM volume control */
+#define   ICE1712_SERR_LEVEL  0x08 /* SERR# level otherwise edge */
+#define   ICE1712_SERR_ASSERT_SB 0x02 /* disabled SERR# assertion for SB irq otherwise enabled */
 #define   ICE1712_NATIVE  0x01 /* native mode otherwise SB */
 #define ICE1712_REG_IRQMASK  0x01 /* byte */
-#define   ICE1712_IRQ_MPU1  0x80
-#define   ICE1712_IRQ_TIMER  0x40
-#define   ICE1712_IRQ_MPU2  0x20
-#define   ICE1712_IRQ_PROPCM  0x10
+#define   ICE1712_IRQ_MPU1  0x80 /* MIDI irq mask */
+#define   ICE1712_IRQ_TIMER  0x40 /* Timer mask */
+#define   ICE1712_IRQ_MPU2  0x20 /* Secondary MIDI irq mask */
+#define   ICE1712_IRQ_PROPCM  0x10 /* professional multi-track */
 #define   ICE1712_IRQ_FM  0x08 /* FM/MIDI - legacy */
 #define   ICE1712_IRQ_PBKDS  0x04 /* playback DS channels */
 #define   ICE1712_IRQ_CONCAP  0x02 /* consumer capture */
-- 
1.7.1